details.js

```diff
@@ -0,0 +1,69 @@
+document.addEventListener("DOMContentLoaded", function() {
+    const countryName = new URLSearchParams(location.search).get('name');
+    const flagImage = document.querySelector(".country-flag img");
+    const countryNameH1 = document.querySelector(".country-flag h1");
+    const nativeName = document.querySelector(".native-name");
+    const population = document.querySelector(".population");
+    const region = document.querySelector(".region");
+    const subRegion = document.querySelector(".sub-region");
+    const capital = document.querySelector(".capital");
+    const topLevelDomain = document.querySelector(".top-level-domain");
+    const currencies = document.querySelector(".currencies");
+    const languages = document.querySelector(".languages");
+
+    fetch(`https://restcountries.com/v3.1/name/${countryName}?fullText=true`)
+        .then(res => res.json())
+        .then(data => {
+            console.log(data);
+            const countryData = data[0]; // Assign data[0] to countryData
+            // Update flag image source
+            flagImage.src = countryData.flags.svg;
+            // Update country name
+            countryNameH1.textContent = countryName;
+            // Update native name if available
+            if (countryData.name && countryData.name.native) {
```

Comment:
Remvoe all the comments, your code should explain itself. 

Also - when you add this kind of comment `Assign data[0] to countryData` you are explainnig how the language is working and its not needed